colcon_package_selection/package_selection/dependencies.py

```diff
@@ -50,6 +55,19 @@ def check_parameters(self, args, pkg_names):  # noqa: D102
                     "Package '{name}' specified with --packages-above "
                     'was not found'
                     .format_map(locals()))
+        if args.packages_above_depth:
+            depth = args.packages_above_depth[0]
+            if not depth.isdigit() or int(depth) < 0:
+                sys.exit(
+                    "Invalid depth '{depth}' specified with "
+                    '--packages-above-depth'
+                    .format_map(locals()))
+            for name in args.packages_above_depth[1:]:
+                if name not in pkg_names:
+                    sys.exit(
+                        "Package '{name}' specified with "
+                        '--packages-above-depth was not found'
+                        .format_map(locals()))
```

Comment:
Instead of deferring all checks until this function is called the `add_argument()` call could pass the `action` keyword argument with a custom action to perform some checking right away:

```
class DepthAndPackageNames(argparse.Action):
    """Action to assign an integer depth optional package names."""

    def __call__(self, parser, namespace, values, option_string=None):
        assert len(values) >= 1
        try:
            values[0] = int(values[0])
            if values[0] < 0:
                raise ValueError()
        except ValueError:
            raise argparse.ArgumentError(
                self, 'the first parameter must be a non-negative integer for '
                'the depth')
        setattr(namespace, self.dest, values)
```

Reply:
Looks nice - thanks.